src/script/util/LocalizerUtil/LocalizerUtil.ts

```diff
@@ -135,14 +162,14 @@ export const setStrings = (newStrings: typeof strings): void => {
   strings = newStrings;
 };
 
-export function t(
-  identifier: StringIdentifer,
-  substitutions?: Substitutes,
-  dangerousSubstitutions?: Record<string, string>,
-  skipEscape: boolean = false,
-): string {
-  return LocalizerUtil.translate(identifier, substitutions, dangerousSubstitutions, skipEscape);
-}
+export const t = <Id extends StringIdentifer>(
+  identifier: Id,
+  ...args: ExtractSubstitutionKeys<TranslationStrings[Id]> extends never
+    ? [substitutions?: undefined, dangerousSubstitutions?: Record<string, string>, skipEscape?: boolean]
+    : [substitutions: Substitutions<Id>, dangerousSubstitutions?: Record<string, string>, skipEscape?: boolean]
```

Comment:
unfortunately, the `args` are mandatory here (instead of typing each parameter separately), because we have to mark `substitutions` either **optional** (the `?` sign) with undefined type, or **required** with the `Substitutions<Id>` type